nautobot_ssot/migrations/0005_sync_compressed_diff.py

```diff
@@ -0,0 +1,34 @@
+# Generated by Django 3.2.16 on 2023-05-23 20:05
+
+from django.db import migrations, models
+import nautobot_ssot.models
+
+
+def copy_diff_to_compressed_diff(apps, schema_editor):
+    for sync in nautobot_ssot.models.Sync.objects.all():
+        sync.compressed_diff = sync.diff
+        sync.validated_save()
+
+
+def undo_diff_copy(apps, schema_editor):
+    pass
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("nautobot_ssot", "0004_sync_summary"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="sync",
+            name="diff",
+            field=models.JSONField(blank=True, null=True),
+        ),
+        migrations.AddField(
+            model_name="sync",
+            name="compressed_diff",
+            field=nautobot_ssot.models.CompressedJSONField(blank=True, default=dict),
+        ),
+        migrations.RunPython(copy_diff_to_compressed_diff, undo_diff_copy),
```

Comment:
https://docs.djangoproject.com/en/3.2/ref/migration-operations/#runpython

> Much like [RunSQL](https://docs.djangoproject.com/en/3.2/ref/migration-operations/#django.db.migrations.operations.RunSQL), ensure that if you change schema inside here you’re either doing it outside the scope of the Django model system (e.g. triggers) or that you use [SeparateDatabaseAndState](https://docs.djangoproject.com/en/3.2/ref/migration-operations/#django.db.migrations.operations.SeparateDatabaseAndState) to add in operations that will reflect your changes to the model state - otherwise, the versioned ORM and the autodetector will stop working correctly.
> 
> By default, RunPython will run its contents inside a transaction on databases that do not support DDL transactions (for example, MySQL and Oracle). This should be safe, but may cause a crash if you attempt to use the schema_editor provided on these backends; in this case, pass atomic=False to the RunPython operation.
> 
> On databases that do support DDL transactions (SQLite and PostgreSQL), RunPython operations do not have any transactions automatically added besides the transactions created for each migration. Thus, on PostgreSQL, for example, you should avoid combining schema changes and RunPython operations in the same migration or you may hit errors like OperationalError: cannot ALTER TABLE "mytable" because it has pending trigger events.